Added navigation for request Body files

mondhs opened the pull request:
Request add navigation for request body files.

Idea is that if I have many request files it would be nice to jump over when ever I need for review or editing.  It would like this:

![test_httpyac-1703946442023](https://github.com/AnWeber/vscode-httpyac/assets/1858749/83282bd0-ff34-42a7-9bae-4c6e6b52dfda)

Currently it works with json extension files as I am using them most, but this filtering list might externalized to configuration in a future.

My code is based on https://github.com/abierbaum/vscode-file-peek

AnWeber commented:
The idea is great. I'm just wondering whether it's really necessary to parse the location links separately. I already parse the file with the httpyac lib to provide the code lens. [To get the needed Positions for the CodeLens](https://github.com/AnWeber/vscode-httpyac/blob/main/src/provider/codeLensProvider.ts#L67) I use a [HttpSymbol
](https://github.com/AnWeber/httpyac/blob/main/src/models/httpSymbol.ts#L3). If we also specifically mark the links in these symbols (new SymbolKind.File), we could generalize the problem and thus also use the parsing for the import of Proto files, for example.

AnWeber commented on src/provider/httpDefinitionProvider.ts:
```diff
@@ -0,0 +1,101 @@
+
+    // if we have an extension, then try replacing it.
+    const parsed_path = path.parse(lookupPath);
+    if (parsed_path.ext !== '') {
```
When does the special case occur that the file extension cannot be determined?

mondhs replied:
I removed this code. Indeed it was redundant.

AnWeber commented on src/provider/httpDefinitionProvider.ts:
```diff
@@ -0,0 +1,101 @@
+    const word = document.getText(document.getWordRangeAtPosition(position));
+    const line = document.lineAt(position);
+
+    this.log_debug('====== peek-file definition lookup ===========', '');
```
There would be too many log_debug outputs that replace a simple breakpoint. Are you having problems with the setup for debugging? Can I help?

mondhs replied:
I just personal preference, I like logs more :). I moved to breakpoints at this moment.

mondhs commented:
I refactored code to used HttpRegions and removed bunch of redundant code and overwhelming logging.

If it would be symbol such as file for requests, imports etc, it would be possible get universal solution.

mondhs commented:
@AnWeber Could I change my pull request to make it acceptable?

AnWeber commented:
@mondhs Hi sorry the ball is in my court. Unfortunately, I'm really out of action at the moment due to a corona illness. I'm slowly getting back. I'll take a look at it soon. Sorry for the delay.

AnWeber commented:
thx for your contribution
